lib/services/local/integrations.go

```diff
@@ -127,22 +127,46 @@ func (s *IntegrationsService) DeleteIntegration(ctx context.Context, name string
 		return trace.Wrap(err)
 	}
 
-	conditionalActions, err := notReferencedByEAS(ctx, s.backend, name)
+	deleteConditions, err := integrationConditionalDeletion(ctx, s.backend, name)
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	conditionalActions = append(conditionalActions, backend.ConditionalAction{
+	deleteConditions = append(deleteConditions, backend.ConditionalAction{
 		Key:       s.svc.MakeKey(backend.NewKey(name)),
 		Condition: backend.Exists(),
 		Action:    backend.Delete(),
 	})
-	_, err = s.backend.AtomicWrite(ctx, conditionalActions)
+	_, err = s.backend.AtomicWrite(ctx, deleteConditions)
 	return trace.Wrap(err)
 }
 
-// notReferencedByEAS returns a slice of ConditionalActions to use with a backend.AtomicWrite to ensure that
+// integrationConditionalDeletion returns a BadParameter error if the integration is referenced by another
+// Teleport service. If it does not find any direct reference, the backend.ConditionalAction is returned
+// with the current state of reference, which should be added to AtomicWrite to ensure that the current
+// reference state remains unchanged until the integration is completeley deleted.
+// Service may have zero or multuple ConditionalActions returned.
+func integrationConditionalDeletion(ctx context.Context, bk backend.Backend, name string) ([]backend.ConditionalAction, error) {
+	var deleteConditionalActions []backend.ConditionalAction
+	easDeleteConditions, err := integrationReferencedByEAS(ctx, bk, name)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	deleteConditionalActions = append(deleteConditionalActions, easDeleteConditions...)
+
+	awsIcDeleteCondition, err := integrationReferencedByAWSICPlugin(ctx, bk, name)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if awsIcDeleteCondition != nil {
```

Comment:
Nit: To keep interfaces of all `integrationReferencedByXXX` methods similar, can we make this one return a list as well so you can do append like above?